sdk/aqueduct/dag.py

```diff
@@ -50,6 +54,36 @@ class Config:
             "operators_by_name": {"exclude": ...},
         }
 
+    def verify(self) -> None:
+        """This method must be called before the DAG is published in publish_flow(), as a final check that
+        everything is consistent.
+
+        Currently, it:
+        1) Makes sure that the specified compute engine can handle the specified resource requests.
+        """
+        allowed_customizable_resources: Dict[str, bool] = {
+            "num_cpus": False,
+            "memory": False,
+        }
+        if self.engine_config.type == RuntimeType.K8S:
+            allowed_customizable_resources = {
+                "num_cpus": True,
+                "memory": True,
+            }
+
+        for op in self.operators.values():
+            if op.spec.resources is not None:
+                if not allowed_customizable_resources["num_cpus"] and op.spec.resources.num_cpus:
+                    raise InvalidUserArgumentException(
+                        "Operator %s cannot configure the number of cpus, since it is not supported when running on %s."
+                        % (op.name, self.engine_config.type)
+                    )
+                if not allowed_customizable_resources["memory"] and op.spec.resources.memory_mb:
+                    raise InvalidUserArgumentException(
+                        "Operator %s cannot configure the amount of memory, since it is not supported when running on %s."
+                        % (op.name, self.engine_config.type)
```

Comment:
Now that I think about it, there's no reason this shouldn't go within `publish_flow()` directly as a helper. This `dag.verify()` stuff is confusing. Will move.